src/transcode/encode.ts

```diff
@@ -0,0 +1,163 @@
+import { BigNumber } from 'ethers';
+import { isBytes, ParamType } from 'ethers/lib/utils';
+import { isBigNumberish } from '@ethersproject/bignumber/lib/bignumber';
+import {
+  isPrimitiveParam,
+  SolValue,
+  toUintOrFelt,
+  safeNext,
+  parseSolAbi,
+  selectSignature,
+  parseParam,
+  Param,
+} from './utils';
+import Web3 from 'web3';
+import { parse } from '../utils/functionSignatureParser';
+
+export async function encodeInputs(
+  filePath: string,
+  func: string,
+  useCairoABI: boolean,
+  rawInputs?: string[],
+): Promise<[string, string]> {
+  if (useCairoABI) {
+    const inputs = rawInputs ? `--inputs ${rawInputs.join(' ').split(',').join(' ')}` : '';
+    return [func, inputs];
+  }
+
+  const solABI = parseSolAbi(filePath);
+  const [funcSignature] = await selectSignature(solABI, func);
+  const selector = new Web3().utils.keccak256(funcSignature).substring(2, 10);
+
+  const funcName = `${func}_${selector}`;
+  const inputs = rawInputs
+    ? `--inputs ${transcodeCalldata(funcSignature, parseParam(rawInputs.join(' ')))
+        .map((i) => i.toString())
+        .join(' ')}`
+    : '';
+
+  return [funcName, inputs];
+}
+
+export function transcodeCalldata(funcSignature: string, inputs: Param): bigint[] {
+  return parse(funcSignature)(inputs);
```

Comment:
In a future PR can you delete the functionSIgnatureParser and use the encode from below and get the ParamType from the ethers library

Reply:
Ok, I will push this change on another PR